modules/metadata-service/api-gateway.tf

```diff
@@ -135,6 +135,17 @@ resource "aws_api_gateway_method_response" "db" {
 resource "aws_api_gateway_deployment" "this" {
   rest_api_id = aws_api_gateway_rest_api.this.id
 
+  # Hacky-ish way to force redeployment whenenver one of the 
+  # relevant aspects that require API deployment change. These are
+  # 1. allowed IP addresses
+  # 2. Change in type of API auth method
+  # 3. Changes in the content of the current file
+  triggers = {
+    cidrs    = join(",", var.access_list_cidr_blocks)
+    auth     = var.api_basic_auth
+    file_md5 = md5(file("${path.module}/api-gateway.tf"))
+  }
+
```

Comment:
I think this won't trigger changes based on changes to other provided tf vars that affect the api right? I guess the only other ones are the `resource_prefix`, `resource_suffix` and `tags` though

Reply:
Yeah but I was thinking changing `resource_prefix` and `resource_suffix` will essentially create a new API Gateway resource which will have naturally gets its own new deployment since the deployment resource will point to a new rest_api_id
```
...
  rest_api_id = aws_api_gateway_rest_api.this.id
...
```